src/EditorFeatures/Core/ValueTracking/SerializableValueTrackedItem.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ValueTracking
+{
+    [DataContract]
+    internal sealed class SerializableValueTrackedItem
+    {
+        [DataMember(Order = 0)]
+        public SerializableSymbolAndProjectId SymbolAndProjectId { get; }
+
+        [DataMember(Order = 1)]
+        public SerializableValueTrackedItem? Parent { get; }
```

Comment:
I'm assuming its fine to have self-referential stuff like this or you wouldn't have done it, but I'm wondering, does anything actually need to look all the way up the tree? Or do things just need to know their immediate parent?

I'm mainly just wondering (out loud) if its more efficient to just serialize the parent `SymbolAndProjectId` rather than the whole parent object.

Reply:
I think symbol isn't sufficient to reconstruct what's needed though. You need to know what document and TextSpan are there as well... 

As far as if this is a good idea? I don't know... it worked ＼（〇_ｏ）／